packages/frontpage/app/api/cron/rank-posts/route.ts

```diff
@@ -0,0 +1,51 @@
+import { db } from "@/lib/db";
+import { sendDiscordMessage } from "@/lib/discord";
+import type { NextRequest } from "next/server";
+import * as schema from "@/lib/schema";
+import { sql } from "drizzle-orm";
+
+export async function GET(request: NextRequest) {
+  const authHeader = request.headers.get("authorization");
+  if (authHeader !== `Bearer ${process.env.CRON_SECRET}`) {
+    await sendDiscordMessage({
+      embeds: [
+        {
+          title: "Unauthorized request to cron endpoint",
+          description: `Request: ${request.url}`,
+          color: 16711680,
+        },
+      ],
+    });
+    return new Response("Unauthorized", {
+      status: 401,
+    });
+  }
+
+  try {
+    await db.update(schema.PostAggregates).set({
+      rank: sql`
+          CAST(COALESCE(${schema.PostAggregates.voteCount}, 1) AS REAL) / (
+            pow(
+              (JULIANDAY('now') - JULIANDAY(${schema.PostAggregates.createdAt})) * 24 + 2,
+              1.8
+            )
+          )
+        `,
```

Comment:
Can we extract and share this sql clause? It's repeated 5 times throughout the app.